modules/jta/src/main/java/org/jboss/weld/jta/TransactionalObserverNotifier.java

```diff
@@ -78,7 +82,22 @@ public ObserverNotifier create(String contextId, TypeSafeObserverResolver resolv
             for (ObserverMethod<? super T> observer : observers) {
                 deferNotification(event, metadata, observer, notifications);
             }
-            transactionServices.registerSynchronization(new TransactionNotificationSynchronization(notifications));
+            try {
+                transactionServices.registerSynchronization(new TransactionNotificationSynchronization(notifications));
+            } catch (Exception e) {
+                if (e.getCause() instanceof RollbackException || e.getCause() instanceof IllegalStateException) {
+                    List<ObserverMethod<? super T>> filteredObservers = observers.stream()
+                            .filter(observerMethod -> !observerMethod.getTransactionPhase().equals(TransactionPhase.AFTER_SUCCESS)).collect(
+                                    Collectors.toList());
+                    filteredObservers.sort((o1, o2) -> {
```

Comment:
Hm, I think this should work: `.filter(observerMethod -> !observerMethod.getTransactionPhase().equals(TransactionPhase.AFTER_SUCCESS))
.sorted((o1, o2) -> o1.getTransactionPhase().compareTo(o2.getTransactionPhase())
.collect(Collectors.toList())`

But I don't like the comparison by ordinals... https://github.com/weld/core/pull/1315#discussion_r58501527